### PR TITLE
feat: cicd workflows for prod env

### PR DESCRIPTION
**Pull request by tommy-ca**
# Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context.

## Checklist before requesting a review

Please delete options that are not relevant.

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented hard-to-understand areas
- [ ] I have ideally added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged

### CODE REVIEW

File: .aws/task_definition.json [hunk 1]
```diff
@@ -29,28 +28,29 @@
             "environment": [],
             "environmentFiles": [
                 {
-                    "value": "arn:aws:s3:::quivr-env-variables/production.env",
+                    "value": "arn:aws:s3:::sr-vt-uw1-prod-env-bucket/production.env",
```
Comment (haohaostayreal) [inline .aws/task_definition.json:31]: use `arn:aws:s3:::sr-vt-env-variables/production.env` as we discussed. Thanks.

File: .aws/task_definition.json [hunk 2]
```diff
@@ -1,14 +1,13 @@
 {
-    "taskDefinitionArn": "arn:aws:ecs:eu-west-3:253053805092:task-definition/quivr:160",
+    "taskDefinitionArn": "arn:aws:ecs:us-west-1:155722108667:task-definition/sr-vt-prod-api:2",
     "containerDefinitions": [
         {
-            "name": "quivr",
-            "image": "253053805092.dkr.ecr.eu-west-3.amazonaws.com/quivr:bada136312ad3497664c3562a36b263d43c89c53",
-            "cpu": "1024",
-            "memory": "2048",
+            "name": "sr-vt-prod-api",
```
Comment (haohaostayreal) [inline .aws/task_definition.json:5]: we decided to call this `sr-vt-prod-backend`? 
`sr-vt-prod-api` is fine, but please purpose the name change easier in the future.

File: .aws/task_definition_prod_worker.json [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
-    "taskDefinitionArn": "arn:aws:ecs:eu-west-3:253053805092:task-definition/quivr-prod-chat:2",
+    "taskDefinitionArn": "arn:aws:ecs:us-west-1:155722108667:task-definition/sr-vt-prod-worker:1",
     "containerDefinitions": [
         {
-            "name": "quivr-chat",
-            "image": "253053805092.dkr.ecr.eu-west-3.amazonaws.com/quivr:85ae06c82935028d828f60f35c7e0a47fdee1ff2",
-            "cpu": "256",
-            "memory": "1024",
+            "name": "sr-vt-prod-worker",
+            "image": "155722108667.dkr.ecr.us-west-1.amazonaws.com/sr-vt-prod-repo",
```
Comment (haohaostayreal) [inline .aws/task_definition_prod_worker.json:6]: I mentioned a couple time, we will use one repo? Please follow our decision.

File: .aws/task_definition_prod_worker.json [hunk 4]
```diff
@@ -18,28 +18,29 @@
...
+                    "awslogs-stream-prefix": "sr-vt-prod-worker"
                 }
             }
         }
     ],
-    "family": "quivr-prod-chat",
-    "taskRoleArn": "arn:aws:iam::253053805092:role/ecsTaskExecutionRole",
-    "executionRoleArn": "arn:aws:iam::253053805092:role/ecsTaskExecutionRole",
+    "family": "sr-vt-prod-worker",
+    "taskRoleArn": "arn:aws:iam::155722108667:role/sr-vt-prod-worker-task",
```
Comment (haohaostayreal) [inline .aws/task_definition_prod_worker.json:40]: can we use arn:aws:iam::155722108667:role/sr-vt-github-actions for now?
do we need to create all these roles as such a early stage of a project?

File: .github/workflows/aws.yml [hunk 5]
```diff
@@ -6,10 +6,10 @@ on:
       - 'v*'
 
 env:
-  AWS_REGION: eu-west-3
-  ECR_REPOSITORY: quivr
-  ECR_REGISTRY: 253053805092.dkr.ecr.eu-west-3.amazonaws.com
-  ECS_CLUSTER: quivr
+  AWS_REGION: us-west-1
+  ECR_REPOSITORY: sr-vt-prod-repo
```
Comment (haohaostayreal) [inline .github/workflows/aws.yml:10]: ECR_REPOSITORY: sr-vt-repo
please

File: .github/workflows/aws.yml [hunk 6]
```diff
@@ -69,14 +71,14 @@ jobs:
       fail-fast: false
       matrix:
         include:
-          - name: "quivr"
-            service: "quivr-backend"
+          - name: "quivr-api"
```
Comment (haohaostayreal) [inline .github/workflows/aws.yml]: no quivr in this file. 
replace with `sr-vt-prod-api`

**Review by haohaostayreal — COMMENTED** [review]
leave some comments.

File: .aws/task_definition.json (hunk 1 above)
Comment (haohaostayreal) [reply .aws/task_definition.json:31]: also cleanup all other buckets please. 
we have ~ 10 buckets now, while we only need like 2-3.

File: .aws/task_definition_prod_worker.json (hunk 4 above)
Comment (haohaostayreal) [reply .aws/task_definition_prod_worker.json:40]: I am fine with having all these roles, if we are following the best practise. 
but I only tested `sr-vt-github-actions` role.
so if we decide to do it differently for production, we have to test the workflow a couple times at least. 
to make sure it works.

File: .aws/task_definition.json (hunk 1 above)
Comment (tommy-ca) [reply .aws/task_definition.json:31]: Even though we are deploying the different stacks in the same vpc/region, we still have to isolate the different set of resources, so we can mange the state with terraform or other IaC tools. So they will remain self contained and can be destroyed.

This is a start for later to deploy to different accounts or regions.

Let us share nothing between envs.

Comment (haohaostayreal) [reply .aws/task_definition.json:31]: Sure. Sounds like a plan. 👍

File: .aws/task_definition_prod_worker.json (hunk 4 above)
Comment (tommy-ca) [reply .aws/task_definition_prod_worker.json:40]: This is following best practices to keep iam roles with least privileges in prod envs.

Later we will revisit the iam role with admin rights for preview env to secure it, so we would not affect the prod env.

File: .github/workflows/aws.yml (hunk 6 above)
Comment (tommy-ca) [reply .github/workflows/aws.yml]: Ok, I will update this.

File: .aws/task_definition.json (hunk 1 above)
Comment (tommy-ca) [reply .aws/task_definition.json:31]: Even though we now share a same region for the preview and prod env, we should keep the resources sets for these envs completely isolated for future update. You could easy remove any resource in a env without affect others. And the deployment IaC code could be designed to treat any architecture as the same, no mater if we share a same region, or share a same account.

**Comment by tommy-ca** [comment]
I have validated the workflow locally with [act](https://github.com/nektos/act), which helps a lot to develop and test workflows. It works as expected.

**Review by haohaostayreal — APPROVED** [review]
Nice

**Pull request merged**